### PR TITLE
Explain how the scroll controller behaves under tests

### PR DESCRIPTION
**Pull request by DaveShuckerow**
## Description

Explain that awaiting ScrollController.animateTo can cause tests to hang.

## Tests

This is a documentation update.

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I signed the [CLA].
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] All existing and new tests are passing.
- [x] The analyzer (`flutter analyze --flutter-repo`) does not report any problems on my PR.
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Did any tests fail when you ran them? Please read [Handling breaking changes].

- [x] No, no existing tests failed, so this is *not* a breaking change.
- [ ] Yes, this is a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Test Coverage]: https://github.com/flutter/flutter/wiki/Test-coverage-for-package%3Aflutter
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[Handling breaking changes]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes

### CODE REVIEW

File: packages/flutter/lib/src/widgets/scroll_controller.dart [hunk 1]
```diff
@@ -142,6 +142,10 @@ class ScrollController extends ChangeNotifier {
   ///
   /// The duration must not be zero. To jump to a particular value without an
   /// animation, use [jumpTo].
+  ///
+  /// Note that when calling [animateTo] in widget tests, `await`ing the
```
Comment (goderbauer) [inline packages/flutter/lib/src/widgets/scroll_controller.dart]: nit: as per style guide don't use "note that".

**Review by goderbauer — APPROVED** [review]
LGTM after style comment is addressed.

File: packages/flutter/lib/src/widgets/scroll_controller.dart (hunk 1 above)
Comment (DaveShuckerow) [reply packages/flutter/lib/src/widgets/scroll_controller.dart]: Done.

**Comment by Piinks** [comment]
It looks like this is experiencing a flake that has been patched. Can you update your branch with changes from master?

**Pull request merged**